code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityMap.java

```diff
@@ -310,14 +308,17 @@ static IdentityMap fromXDMMap(final Map<String, Object> map) {
 		final IdentityMap identityMap = new IdentityMap();
 		for (final String namespace : identityMapDict.keySet()) {
 			try {
-				final ArrayList<HashMap<String, Object>> idArr = (ArrayList<HashMap<String, Object>>) identityMapDict.get(
-					namespace
+				final List<Object> immutableIdList = DataReader.optTypedList(
+					Object.class,
+					identityMapDict,
+					namespace,
+					null
 				);
-				if (idArr == null) {
-					continue;
-				}
+				if (immutableIdList == null) continue;
+
+				final ArrayList<?> idMapList = new ArrayList<>(immutableIdList);
```

Comment:
Note to self - This can be replaced with a direct opt* call when this [change](https://github.com/adobe/aepsdk-core-android/pull/275) is merged.

Reply:
The change was merged and released as part og alpha.4 release. I incorporated it in this PR.